plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -73,46 +73,36 @@ func (c *CachedNodeInfo) GetNodeInfo(id string) (*api.Node, error) {
 }
 
 func isVolumeConflict(volume api.Volume, pod *api.Pod) bool {
-	if volume.GCEPersistentDisk != nil {
-		disk := volume.GCEPersistentDisk
-
-		existingPod := &(pod.Spec)
-		for ix := range existingPod.Volumes {
-			if existingPod.Volumes[ix].GCEPersistentDisk != nil &&
-				existingPod.Volumes[ix].GCEPersistentDisk.PDName == disk.PDName &&
-				!(existingPod.Volumes[ix].GCEPersistentDisk.ReadOnly && disk.ReadOnly) {
+	// fast path if there is no conflict checking targets.
+	if volume.GCEPersistentDisk != nil && volume.AWSElasticBlockStore == nil && volume.RBD == nil {
+		return false
+	}
+
+	for _, existingVolume := range pod.Spec.Volumes {
+		// Same GCE Disk can be mounted as read-only by multiple pod simultaneously.
```

Comment:
Slightly clearer comment: "Same GCE disk mounted by multiple pods conflicts unless all pods mount it read-only."